tests/shared_test_functions.sh

```diff
@@ -108,16 +108,32 @@ has_pid() {
 	return 1
 }
 
-## wait_for_file (filename):
-# Waits until ${filename} exists.
-wait_for_file() {
-	filename=$1
-	while [ ! -e "${filename}" ]; do
+## wait_while(func):
+# Wait while ${func} returns 0.  If ${msleep} is defined, use that to wait
```

Comment:
The current function doesn't return non-zero (it loops infinitely), but I'm leaving the door open for a later version to take a timeout value and return non-zero if that timeout is exceeded.

(The current kivaloo tests have a few "wait X seconds; if Y isn't done, bail with an error" cases.)